test.py

```diff
@@ -0,0 +1,166 @@
+        resource = resource_filename('sme_finance_application_schema', schema_name)
+        with open(resource) as f:
+            schema = json.loads(f.read())
+            validator = jsonschema.validators.Draft4Validator(schema)
+            patch_store(validator.resolver.store)
+            self.assertTrue(validator.is_valid(data))
+
+
+    def test_sample_data_is_valid(self):
+        self.validity_of_data_subtest(SME_V5,'sme_v5')
+        self.validity_of_data_subtest(SME_CONTACT_V3,'sme_contact_v3')
+        self.validity_of_data_subtest(FINANCE_APPLICATION_V3,'finance_application_v3')
+        self.validity_of_data_subtest(FINANCE_NEED_V1,'finance_need_v1')
+        self.validity_of_data_subtest(ENTITY_V1,'entity_v1')
+        self.validity_of_data_subtest(PERSON_V1,'person_v1')
+        self.validity_of_data_subtest(ADDRESS_V1,'address_v1')
+        self.validity_of_data_subtest(ACTOR_V1_DIRECTOR_1,'actor_v1')
+        self.validity_of_data_subtest(ACTOR_V1_DIRECTOR_2,'actor_v1')
+        self.validity_of_data_subtest(ACTOR_V1_GUARANTOR,'actor_v1')
+
+
+    def test_sample_data_is_complete(self):
+        self.completion_of_data_subtest(SME_V5,'sme_v5')
+        self.completion_of_data_subtest(SME_CONTACT_V3,'sme_contact_v3')
+        self.completion_of_data_subtest(FINANCE_APPLICATION_V3,'finance_application_v3')
+        self.completion_of_data_subtest(FINANCE_NEED_V1,'finance_need_v1')
+        self.completion_of_data_subtest(ENTITY_V1,'entity_v1')
+        self.completion_of_data_subtest(PERSON_V1,'person_v1')
+        self.completion_of_data_subtest(ADDRESS_V1,'address_v1')
+        self.completion_of_data_subtest(ACTOR_V1_DIRECTOR_1,'actor_v1')
+        self.completion_of_data_subtest(ACTOR_V1_DIRECTOR_2,'actor_v1')
+        self.completion_of_data_subtest(ACTOR_V1_GUARANTOR,'actor_v1')
+        # TODO - test entity_address and actor_address structures
+
+
+    def test_sme_v5_and_contact_v3_to_finance_application_v3_translator(self):
+        expected_finance_application_v3 = copy.deepcopy(FINANCE_APPLICATION_V3)
+        for field in UNTRANSLATED_FINANCE_APPLICATION_V3_FIELDS:
+            expected_finance_application_v3.pop(field)
+        for field in UNTRANSLATED_ENTITY_V1_FIELDS:
+            expected_finance_application_v3['requesting_entity'].pop(field)
+        for field in UNTRANSLATED_PERSON_V1_FIELDS:
+            expected_finance_application_v3['applicant'].pop(field)
+        for field in UNTRANSLATED_FINANCE_NEED_V1_FIELDS:
+            expected_finance_application_v3['finance_need'].pop(field)
+        for field in UNTRANSLATED_ADDRESS_V1_FIELDS:
+            expected_finance_application_v3['applicant']['addresses'][0]['address'].pop(field)
+
+        translated_finance_application_v3 = sme_v5_and_contact_v3_to_finance_application_v3_translator(SME_V5, SME_CONTACT_V3)
+        self.assertDictEqual(translated_finance_application_v3, expected_finance_application_v3)
+
```

Comment:
Why not use `validity_of_data_subtest(translated_finance_application_v3)`?

Reply:
the translated_finance_application_v3 could be *valid* but *missing fields* if one of the translations fails.